presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -863,7 +870,7 @@ public PlanWithProperties visitSemiJoin(SemiJoinNode node, PreferredProperties p
                     }
                     else {
                         source = withDerivedProperties(
-                                partitionedExchange(idAllocator.getNextId(), REMOTE, source.getNode(), createPartitioning(sourceSymbols), Optional.empty()),
+                                partitionedExchange(idAllocator.getNextId(), getExchangeScope(source.getNode()), source.getNode(), createPartitioning(sourceSymbols), Optional.empty()),
                                 source.getProperties());
                         filteringSource = withDerivedProperties(
                                 partitionedExchange(idAllocator.getNextId(), REMOTE, filteringSource.getNode(), createPartitioning(filteringSourceSymbols), Optional.empty(), true),
```

Comment:
I personally find it difficult to reason/follow why some remote partitioned exchange will call `getExchangeScope`, why some are hard coded to `REMOTE`. This might also make future code maintenance difficult. 

My understanding is we didn't enable it other than JOIN/AGGREGATE. One thought is to pass the "parentNode" to `getExchangeScope`.

Let's discuss in person once @rschlussel  also took a look

Reply:
https://github.com/prestodb/presto/pull/12469#issuecomment-478719893